CC/src/DistanceComputationTools.cpp

```diff
@@ -1174,23 +1173,25 @@ void cloudMeshDistCellFunc_MT(const DgmOctree::IndexAndCode& desc)
 	size_t trianglesToTestCapacity = 0;
 
 	//bit mask for efficient comparisons
-	QBitArray* bitArray = nullptr;
+	std::vector<bool>* bitArray = nullptr;
 	if (s_useBitArrays_MT)
 	{
-		s_currentBitMaskMutex.lock();
 		if (s_bitArrayPool_MT.empty())
 		{
-			bitArray = new QBitArray();
+			bitArray = new std::vector<bool>;
 			bitArray->resize(s_intersection_MT->mesh->size());
-			//s_bitArrayPool_MT.push_back(bitArray);
+			//s_bitArrayPool_MT.push_back(bitArray); //TODO: RJ it's no used because it is added at the end of the process. but why?
+			//is it more efficient?
 		}
 		else
 		{
+			s_currentBitMaskMutex.lock(); //TODO: RJ: use a parallel_queue to avoid implicit lock
 			bitArray = s_bitArrayPool_MT.back();
 			s_bitArrayPool_MT.pop_back();
+			s_currentBitMaskMutex.unlock();
+
 		}
-		s_currentBitMaskMutex.unlock();
-		bitArray->fill(0);
+		std::fill(std::begin(*bitArray), std::end(*bitArray), 0);
```

Comment:
here is the biggest performance bottleneck...

Reply:
(See comment above first)

You might try `(*bitArray).assign(bitArray->size(), false);` instead of `std::fill`, but we're getting into micro-optimizations that are perhaps not useful because they will depend on the compiler/architecture, etc..